Cargo.toml

```diff
@@ -1,21 +1,23 @@
 [package]
 name = "atomic-pool"
-version = "1.0.1"
+version = "2.0.0"
 authors = ["Dario Nieuwenhuis <dirbaio@dirbaio.net>"]
 description = "Statically allocated pool providing a std-like Box."
 repository = "https://github.com/embassy-rs/atomic-pool"
 edition = "2021"
 readme = "README.md"
 license = "MIT OR Apache-2.0"
-categories = [
-    "embedded",
-    "no-std",
-    "concurrency",
-    "memory-management",
+categories = ["embedded", "no-std", "concurrency", "memory-management"]
+
+[features]
+default = ["portable-atomic-critical-section"]
```

Comment:
don't make this default. all it takes is one lib using `atomic-pool` and forgetting `default-features=false` and the user can't use `unsafe-assume-single-core` anymore.

Reply:
The library doesn't compile without at least one of those enabled. Should I default to the other one or inject a compilation error if none are enabled?